stl/inc/xlocnum

```diff
@@ -1292,18 +1292,23 @@ protected:
         _OutIt _Dest, ios_base& _Iosbase, _Elem _Fill, double _Val) const { // put formatted double to _Dest
         string _Buf;
         char _Fmt[8];
-        bool _Isfixed         = (_Iosbase.flags() & ios_base::floatfield) == ios_base::fixed;
-        streamsize _Precision = _Iosbase.precision() <= 0 && !_Isfixed ? 6 : _Iosbase.precision(); // desired precision
-        size_t _Bufsize       = static_cast<size_t>(_Precision);
+        const bool _Isfixed = (_Iosbase.flags() & ios_base::floatfield) == ios_base::fixed;
+        const bool _Ishex   = (_Iosbase.flags() & ios_base::floatfield) == (ios_base::fixed | ios_base::scientific);
+        const streamsize _Precision =
+            _Ishex ? (DBL_MANT_DIG + 2) / 4

```

Comment:
Where does this `(DBL_MANT_DIG + 2) / 4` constant come from?

Reply:
We finally realized the meaning of this calculation - it is subtracting 1 (because `DBL_MANT_DIG` is 53 which includes the implicit bit, subtracting 1 produces 52 which is the number of explicit bits), then adding 3 and dividing by 4 to perform a ceiling-division. This is probably worth a comment.